backend/src/main/kotlin/org/loculus/backend/service/submission/SubmissionDatabaseService.kt

```diff
@@ -984,18 +987,34 @@ class SubmissionDatabaseService(
             Clock.System.now().toEpochMilliseconds() - timeToStaleInSeconds * 1000,
         ).toLocalDateTime(TimeZone.UTC)
 
-        val numberDeleted = SequenceEntriesPreprocessedDataTable.deleteWhere {
-            statusIs(IN_PROCESSING) and startedProcessingAtColumn.less(staleDateTime)
+        transaction {
+            // Check if there are any stale sequences before attempting to delete
+            val staleSequencesExist = SequenceEntriesPreprocessedDataTable
+                .selectAll()
```

Comment:
Blocking: This is equivalent to a `select *`. I guess you don't want to fetch the processed data, including the sequences. Please restrict this to the columns that you actually need.

(The unit tests log the executed SQL statements)

Reply:
As this is a single SQL statement (I would hope) doesn't it optimize away the unnecessary columns?

Reply:
What you are suggesting would result in duplication of the needed columns which might be necessary but only if the ORM exposed is bad 🙃

Reply:
No, how should it know the necessary columns? After `.limit(1)`, you have an iterator over the results. `.empty` is not part of the exposed code anymore (AFAIK), so the generated SQL statement has to be general enough to support anything else, too.

Reply:
Ah I see, I would have hoped/expected the empty is part of the ORM. We need to look at emitted SQL to see!

Reply:
Maybe there is an alternative that is part of Exposed? Counting?